Networking-HOWTO.md

```diff
@@ -1,76 +1,74 @@
 
-You can setup your wireless router, or you can also put your mobile device
-in "Wireless tethering" mode. Note that Kanux fully supports the wireless dongle
-provided. Other WiFi dongles should work as well but have not been fully tested.
+### WiFi
 
-If you need to connect to other wireless networks, open the "Wifi" app icon
-in the Extras folder, or execute "sudo kano-wifi" from the command line.
+Raspbian `Jessie` is not following the usual `Debian` approach to networking with`/etc/network/if-*` scripts. Instead `dhcpcd` is used. `dhcpcd` is an implementation of the `DHCP` client specified in `RFC 2131`. It gets the host information (`IP`address, routes, etc) from a `DHCP` server and configures the network interface of the machine on which it is running.
 
-Once you successfully connect to a wireless network, it will be remembered,
-so next time you boot Kanux it will automatically connect to the network
-if it is in range.
+It then runs the configuration script which writes `DNS` information to `resolvconf`, if available, otherwise directly to `/etc/resolv.conf`. If the `hostname` is currently `blank`, (`null`) or `localhost`, or `force_hostname` is `YES` or `TRUE` or `1` then dhcpcd sets the hostname to the one supplied by the `DHCP` server.
 
-In case you need to fine tune more specific wireless secured networks,
-kano-wifi allows you to provide a custom wpa_supplicant configuration file,
-like this:
+`dhcpcd` then daemonises and waits for the lease renewal time to lapse. It will attempt to renew its lease and reconfigure if the new lease changes, when the lease begins to expire, or the `DHCP` server sends message to renew early.
 
- * sudo kano-wifi /path/to/my/wpa_supplicant.conf
+It turns out that `dhcpcd` has a very neat way of hooking up into the network configuration: all you need to do is drop a script in the `/lib/dhcpcd/dhcpcd-hooks` directory, and it will get called at the various stages of the IP allocation process. The script will receive a bunch of useful variables like `${interface}` and `${new_ip_address}` that make it easy to get all the data we need. Check also the `dhcpcd-run-hooks` man page!
 
-Make sure you provide an absolute path filename to avoid problems during
-automatic connect at boot time. As an example here's a small simple file
-to connect to a WPA2 network:
+#### Graphical
+
+If you need to connect to other wireless networks, open the "Wifi" app icon on the right hand end of the menu bar, or execute "sudo kano-wifi" from the command line. Once you successfully connect to a wireless network, it will be remembered, so next time you boot Kanux it will automatically connect to the network if it is in range.
+
+#### Command Line
+
+In case you need to fine tune more specific wireless secured networks, kano-wifi allows you to provide a custom wpa_supplicant configuration file, like this:
+
+`sudo kano-wifi /path/to/my/wpa_supplicant.conf`
```

Comment:
We should probably remove that script from the project. It's fairly outdated and unmaintained.